internal/cliwrapper/cliwrapper.go

```diff
@@ -9,28 +9,58 @@ import (
 	"io"
 	"os"
 	"os/exec"
+	"path/filepath"
 	"regexp"
 	"strings"
+	"sync"
 )
 
 type cliWrapper struct {
 	pythonFullPath string
-	workDir        string
+	pluginDir      string
+	connectorDir   string
 	deployCommand  *exec.Cmd
 	logger         log.Logger
-	stdErrBuff     bytes.Buffer
+	stdErrBuff     bufferThreadSafe
+}
+
+type bufferThreadSafe struct {
+	b    bytes.Buffer
+	lock *sync.Mutex
+}
+
+func (b *bufferThreadSafe) Len() int {
+	b.lock.Lock()
+	defer b.lock.Unlock()
+	return b.b.Len()
+}
+
+func (b *bufferThreadSafe) Write(p []byte) (int, error) {
+	b.lock.Lock()
+	defer b.lock.Unlock()
+	return b.b.Write(p)
+}
+
+func (b *bufferThreadSafe) String() string {
+	b.lock.Lock()
+	defer b.lock.Unlock()
+	return b.b.String()
 }
```

Comment:
From the source, it looks like `String()` is kind of like syntactical sugar for a type conversion of the bytes buffer at the current offset, and [Read(p []byte)](https://cs.opensource.google/go/go/+/refs/tags/go1.21.5:src/bytes/buffer.go;drc=d8d22664c65b0054fc9ea6ba7e945481348fc8fc;l=318) creates a copy.

Reply:
> So, `Read()` seems like the right thing, while, because you're releasing the lock before the return, `String()` seems like the wrong thing.

Upon review, `String()` returns the contents of the bytes-slice as a `string` which, I believe, results in a copy...so I think it's fine.